scripts/tests/twister/test_harness.py

```diff
@@ -1209,6 +1221,124 @@ def test_gtest_repeated_run(gtest):
+    assert cpputest.status == TwisterStatus.NONE
+
+
+def test_cpputest_start_test(cpputest):
+    process_logs(
+        cpputest,
+        [
+            SAMPLE_CPPUTEST_START_FMT.format(
+                suite="suite_name", test="test_name"
+            ),
+        ],
+    )
+    assert cpputest.status == TwisterStatus.NONE
+    assert len(cpputest.detected_suite_names) == 1
+    assert cpputest.detected_suite_names[0] == "suite_name"
+    assert cpputest.instance.get_case_by_name("id.suite_name.test_name") is not None
+    assert (
+        cpputest.instance.get_case_by_name("id.suite_name.test_name").status == TwisterStatus.STARTED
+    )
+
+
+def test_cpputest_one_test_passed(cpputest):
+    process_logs(
+        cpputest,
+        [
+            SAMPLE_CPPUTEST_START_FMT.format(
+                suite="suite_name", test="test_name"
+            ),
+            SAMPLE_CPPUTEST_END_PASS_FMT.format(
+                tests=1, ran=1, checks=5, ignored=0, filtered=0, time=10
+            )
+        ],
+    )
+    assert len(cpputest.detected_suite_names) == 1
+    assert cpputest.detected_suite_names[0] == "suite_name"
+    assert cpputest.instance.get_case_by_name("id.suite_name.test_name") != TwisterStatus.NONE
+    assert cpputest.instance.get_case_by_name("id.suite_name.test_name").status == TwisterStatus.PASS
+
+
+def test_cpputest_multiple_test_passed(cpputest):
+    logs = []
+    total_passed_tests = 5
+    for i in range(0, total_passed_tests):
+        logs.append(SAMPLE_CPPUTEST_START_FMT.format(suite="suite_name",
+                                                     test=f"test_name_{i}"))
+    logs.append(SAMPLE_CPPUTEST_END_PASS_FMT.format(
+        tests=total_passed_tests, ran=total_passed_tests, checks=5, ignored=0, filtered=0, time=10
+    ))
+    process_logs(cpputest, logs)
+    assert len(cpputest.detected_suite_names) == 1
+    assert cpputest.detected_suite_names[0] == "suite_name"
+    for i in range(0, total_passed_tests):
+        test_name = f"id.suite_name.test_name_{i}"
+        assert cpputest.instance.get_case_by_name(test_name) != TwisterStatus.NONE
+        assert cpputest.instance.get_case_by_name(test_name).status == TwisterStatus.PASS
+
+
+def test_cpputest_test_failed(cpputest):
+    process_logs(
+        cpputest,
+        [
+            SAMPLE_CPPUTEST_START_FMT.format(
+                suite="suite_name", test="test_name"
+            ),
+            SAMPLE_CPPUTEST_FAIL_FMT.format(
+                suite="suite_name", test="test_name"
+            )
+        ],
+    )
+    assert cpputest.status == TwisterStatus.NONE
+    assert len(cpputest.detected_suite_names) == 1
+    assert cpputest.detected_suite_names[0] == "suite_name"
+    assert cpputest.instance.get_case_by_name("id.suite_name.test_name") != TwisterStatus.NONE
+    assert cpputest.instance.get_case_by_name("id.suite_name.test_name").status == TwisterStatus.FAIL
+
+
+def test_cpputest_test_repeated(cpputest):
```

Comment:
What if I want to call `RunAllTests` , multiple times?

Reply:
Oops, I probably wrote it. Give me a bit, I'll take a look and get back to you later today.

Reply:
Marking as resolved, sorry it took so long.